cmd/buildah/config.go

```diff
@@ -193,9 +193,16 @@ func updateConfig(builder *buildah.Builder, c *cobra.Command, iopts configResult
 					return name
 				}
 				env[1] = os.Expand(env[1], getenv)
-				builder.SetEnv(env[0], env[1])
-			} else {
+				if strings.HasSuffix(env[1], "-") {
```

Comment:
This should be env[0] not env[1]

buildah config --env foo- ...

But actually I don't think you need this at all.
Users will not do --foo=bar-